src/state/Store.provider.jsx

```diff
@@ -0,0 +1,22 @@
+import React, { useContext, useReducer } from 'react';
```

Comment:
**Optional**:

You can move all the places in which you have `dispatch` to live inside this file. 

You would then need to change `value={[store, dispatch]}` to be something like `value={{store, setSearchTerm, setTheme}}` that way you can use it like a normal function on the other files